go/login/server/keymanager.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"fmt"
+	"sync"
+
+	"../../glome"
+)
+
+// ErrInvalidKeyIndex denotes that provided index is invalid for a key. Only
+// indexes in range 0-127 are valid.
+type ErrInvalidKeyIndex struct {
+	Index uint8
+}
+
+func (e ErrInvalidKeyIndex) Error() string {
+	return fmt.Sprintf("key index should be in range 0-127, found: %v", e.Index)
+}
+
+// ErrOverloadedKeyIndex denotes that provided index is already in use, so no
```

Comment:
I'd use "Duplicate" instead of "Overloaded".

Reply:
Changed